src/ffi/schema.rs

```diff
@@ -356,6 +356,8 @@ fn to_format(data_type: &DataType) -> String {
             r
         }
         DataType::Dictionary(index, _) => to_format(index.as_ref()),
+        //TODO: get format from metadata
+        DataType::Extension(ty) => ty.to_format().to_string(),
```

Comment:
The extension type does not have a format for FFI: that information is passed on the fields' metadata in the keys `ARROW::extension::name` and `ARROW::extension::metadata`.